src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,57 @@
+/* eslint-disable no-useless-constructor */
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  static removeDead(animal) {
+    Animal.alive = Animal.alive.filter((a) => a !== animal);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, hidden = false, health = 100) {
+    super(name, health);
+    this.hidden = hidden;
+
+    if (this.health <= 0) {
+      Animal.removeDead(this);
+    }
+  }
+
+  hide() {
+    this.hidden = true;
+  }
+
+  takeDamage(amount) {
+    this.health -= amount;
+
+    if (this.health <= 0) {
+      Animal.removeDead(this);
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(target) {
+    if (target instanceof Herbivore) {
+      if (!target.hidden) {
+        target.takeDamage(50);
+      } else {
+      }
+    } else {
+    }
```

Comment:
The `else` block here is empty. Consider adding logic or a comment to handle cases where the target is not a `Herbivore`.